packages/gatsby-plugin-sharp/src/process-file.js

```diff
@@ -75,25 +88,50 @@ exports.processFile = (file, contentDigest, transforms, options = {}) => {
     // this is an experimental api so it can be removed without any warnings
     if (process.env.GATSBY_CLOUD_IMAGE_SERVICE_URL) {
       let cloudPromise
+      const cloudJobID = uuidv4()
 
       return transforms.map(transform => {
         if (!cloudPromise) {
           cloudPromise = got
             .post(process.env.GATSBY_CLOUD_IMAGE_SERVICE_URL, {
               body: {
+                id: cloudJobID,
                 file,
                 hash: contentDigest,
                 transforms,
                 options,
               },
               json: true,
             })
+            .then(
+              () =>
+                new Promise((resolve, reject) => {
+                  console.log(`Sent request ${cloudJobID} / ${file}`)
+                  const eventToListenFor = `GATSBY_CLOUD_IMAGE_SERVICE_${cloudJobID}`
+                  const handler = msg => {
+                    cloudEventsEmitter.off(eventToListenFor, handler)
+                    console.log(
+                      `Got finish event ${cloudJobID} / ${file} - ${
+                        msg.err ? `ERROR` : `SUCCESS`
+                      }`
+                    )
+                    if (msg.err) {
+                      reject()
+                    } else {
+                      resolve()
+                    }
+                  }
+
+                  cloudEventsEmitter.on(eventToListenFor, handler)
+                })
+            )
             .then(() => transform)
 
           return cloudPromise
         }
 
-        return Promise.resolve(transform)
+        // actually wait for cloud promise so reporting is correct
+        return cloudPromise.then(() => transform)
```

Comment:
did this change, because if job has 6 images to produce - it would immediately say that 5 of them are done while actually waiting for all of them